public/app/features/dashboard/state/settings/reducer.ts

```diff
@@ -0,0 +1,23 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { DashboardSettingsState, DashboardVariablesSettings } from './types';
+
+export const initialDashboardSettingsState: DashboardSettingsState = {
+  variables: {
+    showUnknowns: true,
+  },
+};
+
+const dashboardSettingsSlice = createSlice({
+  name: 'dashboard/settings',
+  initialState: initialDashboardSettingsState,
+  reducers: {
+    setVariableSettings: (state, action: PayloadAction<DashboardVariablesSettings>) => {
+      state.variables = action.payload;
```

Comment:
if this lived in the main dashboard slice & reducer this could do: 
state.getModel().setVariableSettings(state.variables); 
That way we could skip the thunk action. maybe a bit hackier to modify the state in the reducer, but not sure if there is a difference in mutating state in a thunk or a reducer. and would reduce the code by a lot :) 

And it would be temporary until we find a way to sidestep DashboardModel

Reply:
Hmm, reducers should never have side effects, for that we have thunks.